datasketch/lsh.py

```diff
@@ -282,6 +301,38 @@ def _insert(
         for H, hashtable in zip(Hs, self.hashtables):
             hashtable.insert(H, key, buffer=buffer)
 
+    def __equivalent(self, other:MinHashLSH) -> bool:
+        """
+        Returns:
+            bool: If the two MinHashLSH has equal num_perm, band size and size of each bands then two are equivalent.
+        """
+        return (
+            type(self) is type(other) and
+            self.h == other.h and
+            self.b == other.b and
+            self.r == other.r
+        )
```

Comment:
There are also:
1. the hash function
2. the weights

Reply:
alright! added it as well.

Reply:
Hmm consider this for function equivalence?
https://stackoverflow.com/questions/20059011/check-if-two-python-functions-are-equal


 Probably a good idea to add some tests for some error cases.

Reply:
I have checked out the link for function equivalence. It was too vague to follow. And I am not sure if we can use the inspect to use the source code of the function to compare. I have tried using two different hash functions one was the in-buit hash method and other md5 hash and sha1. And it worked properly for the normal storage option, but when I tested it for redis I was getting an error since redis key is in bytes and the hash function we are passing. Sharing the snippet. 

`from datasketch.hashfunc import sha1_hash32
lsh3 = MinHashLSH(threshold=0.5, num_perm=128, storage_config={
    'type': 'redis', 'redis': {'host': 'localhost', 'port': 6379}
}, hashfunc=sha1_hash32)`
while trying to insert the minhash object I was getting the TypeError saying **can't concat _hashlib.HASH to bytes** 
Can you please help out here? @ekzhu

Reply:
I think what you have done is great already. If we don't have a precise definition of hash function equivalence, we just don't add the check to make the API well-defined.

Reply:
Regarding the issue with hashfunc, the hash function should input bytes and output bytes. The built-in `sha1_hash32` inputs bytes and output integer -- it is used for `MinHash`, not for `MinHashLSH`.

Reply:
We can remove it and I can do a bit of cleaning and can push the changes but can you help me understanding since we would be facing this TypeError regardless. So, if we are trying to insert any MinHash object into a MinHashLSH object with a hash function defined and the storage options is redis we should be facing it. Or, is there anything that I am doing wrong? I will push the changes today. Thanks @ekzhu

Reply:
BTW, sorry for the delayed response...

Reply:
No issues at all @ekzhu. Just to understand it we can only use the hash functions that either outputs the hash in bytes or a custom one to do the same. Please correct me if I am wrong.

Reply:
You should not have `TypeError`. The default hash function for `MinHashLSH` should work. `sha1_hash32` is for `MinHash` not for `MinHashLSH`. 

`MinHash`'s hash function is not used in `MinHashLSH`.

Reply:
Okay, got it. I will push the changes once done today.